omegaconf/base.py

```diff
@@ -32,7 +33,7 @@
 
 DictKeyType = Union[str, int, Enum, float, bool]
 
-_MARKER_ = object()
+_MARKER_: Any = Marker("_MARKER_")
```

Comment:
Looks like all markers are serving the exact same purpose (act as a sort of NO_DEFAULT_PROVIDED indicator).
We can have a pass later to consolidate them somehow and use a single one.

Reply:
Oh well, I thought why not do it at the same time, one less thing to worry about: see 8862343